correlation_study.py

```diff
@@ -63,16 +63,35 @@ def get_correlated_stocks(self, use_pct_change=False):
             for j in range(len(self.tickers)):
                 if use_pct_change:
                     # Calculate percentage change
-                    vals_i = self.dataframe[self.tickers[i]].pct_change().dropna().to_numpy()
-                    vals_j = self.dataframe[self.tickers[j]].pct_change().dropna().to_numpy()
+                    vals_i = self.dataframe.df[self.tickers[i]].pct_change().dropna().to_numpy()
+                    vals_j = self.dataframe.df[self.tickers[j]].pct_change().dropna().to_numpy()
                 else:
                     # Use original values
-                    vals_i = self.dataframe[self.tickers[i]].to_numpy()
-                    vals_j = self.dataframe[self.tickers[j]].to_numpy()
-                r_ij, p_ij = ss.stats.pearsonr(vals_i, vals_j)
-                corr_values[i, j] = r_ij
-                pvalue_array[i, j] = p_ij
-                
+                    vals_i = self.dataframe.df[self.tickers[i]].to_numpy()
+                    vals_j = self.dataframe.df[self.tickers[j]].to_numpy()
+                # Ensure values are numeric
+                try:
+                    vals_i = pd.to_numeric(vals_i, errors='coerce')
+                    vals_j = pd.to_numeric(vals_j, errors='coerce')
+                    # Filter out NaN values caused by non-numeric data
+                    valid_indices = ~np.isnan(vals_i) & ~np.isnan(vals_j)
+                    vals_i = vals_i[valid_indices]
+                    vals_j = vals_j[valid_indices]
+                    # Check if there's enough valid data to calculate correlation
+                    if len(vals_i) == 0 or len(vals_j) == 0:
+                        corr_values[i, j] = np.nan
+                        pvalue_array[i, j] = np.nan
+                        continue
+
```

Comment:
Isn't it redundant? The tickers should already be cleaned. If there is a reason please write me so I can understand the code better : )

Reply:
This allow us to test the module also with uncleaned datasets (for now we have only uncleaned dataset, the clean one do not work for other issues in the cleaning module). So it's true that cleaning and checking lines are redundant, but we need them for now in my opinion. Otherwise we can't even run the project to see improvements.

Reply:
Oks, for me it's good then. Btw don't close the conversations so when Giulia makes the final checks she can see the various changes. I did this pull request with her yesterday. 
Thanks